src/Components/Blazor/WebAssembly.Authentication/src/WebAssemblyAuthenticationServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Components.Authorization;
+using Microsoft.AspNetCore.Components.WebAssembly.Authentication;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Contains extension methods to add authentication to Blazor webassembly applications.
+    /// </summary>
+    public static class WebAssemblyAuthenticationServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds support for authentication for SPA applications using the given <typeparamref name="TProviderOptions"/> and
+        /// <typeparamref name="TRemoteAuthenticationState"/>.
+        /// </summary>
+        /// <typeparam name="TRemoteAuthenticationState">The state to be persisted across authentication operations.</typeparam>
+        /// <typeparam name="TProviderOptions">The configuration options of the underlying provider being used for handling the authentication operations.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> where the services were registered.</returns>
+        public static IServiceCollection AddSpaAuthentication<TRemoteAuthenticationState, TProviderOptions>(this IServiceCollection services)
```

Comment:
The naming bothers me a bit... The application itself is a SPA, so any authentication you do is SPA authentication by definition. Instead, should this be called "remote" authentication or similar to give it a distinct character of its own?

Let's not be limited by the API names chosen for server-side ASP.NET Core (and even less so, for the other SPA templates) as those are different scenarios.

Reply:
This was renamed to AddRemoteAuthentication